interface/ceed-vector.c

```diff
@@ -695,6 +735,83 @@ int CeedVectorAXPY(CeedVector y, CeedScalar alpha, CeedVector x) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Compute y = alpha x + beta y
+
+  @param[in,out] y     target vector for sum
+  @param[in]     alpha first scaling factor
+  @param[in]     beta  second scaling factor
+  @param[in]     x     second vector, must be different than y
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedVectorAXPBY(CeedVector y, CeedScalar alpha, CeedScalar beta, CeedVector x) {
+  CeedScalar       *y_array = NULL;
+  CeedScalar const *x_array = NULL;
+  CeedSize          n_x, n_y;
+
+  CeedCall(CeedVectorGetLength(y, &n_y));
+  CeedCall(CeedVectorGetLength(x, &n_x));
+  if (n_x != n_y) {
+    // LCOV_EXCL_START
+    return CeedError(y->ceed, CEED_ERROR_UNSUPPORTED, "Cannot add vector of different lengths");
+    // LCOV_EXCL_STOP
+  }
+  if (x == y) {
+    // LCOV_EXCL_START
+    return CeedError(y->ceed, CEED_ERROR_UNSUPPORTED, "Cannot use same vector for x and y in CeedVectorAXPBY");
+    // LCOV_EXCL_STOP
+  }
+
+  bool has_valid_array_x = true, has_valid_array_y = true;
+  CeedCall(CeedVectorHasValidArray(x, &has_valid_array_x));
+  if (!has_valid_array_x) {
+    // LCOV_EXCL_START
+    return CeedError(x->ceed, CEED_ERROR_BACKEND, "CeedVector x has no valid data, must set data with CeedVectorSetValue or CeedVectorSetArray");
+    // LCOV_EXCL_STOP
+  }
+  CeedCall(CeedVectorHasValidArray(y, &has_valid_array_y));
+  if (!has_valid_array_y) {
+    // LCOV_EXCL_START
+    return CeedError(y->ceed, CEED_ERROR_BACKEND, "CeedVector y has no valid data, must set data with CeedVectorSetValue or CeedVectorSetArray");
+    // LCOV_EXCL_STOP
+  }
+
+  Ceed ceed_parent_x, ceed_parent_y;
+  CeedCall(CeedGetParent(x->ceed, &ceed_parent_x));
+  CeedCall(CeedGetParent(y->ceed, &ceed_parent_y));
+  if (ceed_parent_x != ceed_parent_y) {
+    // LCOV_EXCL_START
+    return CeedError(y->ceed, CEED_ERROR_INCOMPATIBLE, "Vectors x and y must be created by the same Ceed context");
+    // LCOV_EXCL_STOP
+  }
+
+  // Backend implementation
+  if (y->AXPBY) {
+    CeedCall(y->AXPBY(y, alpha, beta, x));
+    return CEED_ERROR_SUCCESS;
+  }
+
+  // Default implementation
+  CeedCall(CeedVectorGetArray(y, CEED_MEM_HOST, &y_array));
+  CeedCall(CeedVectorGetArrayRead(x, CEED_MEM_HOST, &x_array));
+
+  assert(x_array);
+  assert(y_array);
+
+  for (CeedInt i = 0; i < n_y; i++) {
+    y_array[i] = beta * y_array[i];
+    y_array[i] += alpha * x_array[i];
+  }
```

Comment:
Why not all at once like
```suggestion
  for (CeedInt i = 0; i < n_y; i++) y_array[i] = alpha * x_array[i] + beta * y_array[i];
```
so it matches the docstring more closely?